enhancements/update/synchronized-upgrades.md

```diff
@@ -0,0 +1,302 @@
+    might be useful?
+
+### Open Questions
+
+- How exactly will the coordination between the two MCO upgrades
+  occur?
+
+- How exactly will infra nodes detect their tenant node rebooting?
+
+- Do we want to allow (z-stream) upgrades of the tenant cluster
+  without also upgrading the infra cluster?
+
+### Risks and Mitigations
+
+The infra cluster could remain stuck at the pre-reboot stage of the
+upgrade for arbitrarily long. In theory this should not cause any
+problems, as all operators must be able to handle running with the old
+RHCOS version anyway. But we should make sure that alerts eventually
+get fired in this case.
+
+More TBD
+
+### Test Plan
+
+TBD
+
+### Graduation Criteria
+
+TBD
+
+#### Dev Preview -> Tech Preview
+
+#### Tech Preview -> GA
+
+#### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+This is a modification to the upgrade process, not something that can
+be upgrade or downgraded on its own.
+
+TBD, as the details depend on the eventual design.
+
+### Version Skew Strategy
+
+TBD, as the details depend on the eventual design.
+
+### Operational Aspects of API Extensions
+
+TBD
+
+#### Failure Modes
+
+- The system might get confused and spuriously block upgrades that
+  should be allowed.
+
+- Communications failures might lead to upgrades failing without the
+  tenant cluster being able to figure out why they failed.
+
+- TBD
+
+#### Support Procedures
+
+TBD
+
+## Implementation History
+
+- Initial proposal: 2021-01-11
+- Updated for initial feedback: 2021-01-24
+
+## Drawbacks
+
+This makes the upgrade process more complicated, which risks rendering
+clusters un-upgradeable without manual intervention.
+
+However, without some form of synchronization, it is impossible to
+have non-disruptive tenant cluster upgrades.
+
+## Alternatives
+
+### Never Reboot the DPUs
+
+This implies never upgrading OCP on the DPUs. I don't see how this
+could work.
+
+### Don't Have an Infra Cluster
+
+If the DPUs were not all part of a single OCP cluster (for example,
+they were just "bare" RHCOS hosts, or they were each running
+Single-Node OpenShift), then it might be simpler to synchronize the
+DPU upgrades with the tenant upgrades, because then each tenant could
+coordinate the actions of its own DPU by itself.
+
+The big problem with this is that, for security reasons, we don't want
+the tenants to have any control over their DPUs. (For some future use
+cases, the DPUs will be used to enforce security policies on their
+tenants.)
+
+### More closely coordinated reboots
+
+In the original proposal, the infra and tenant reboots were more
+closely coordinated, with the infra and tenant MCOs communicating so
+that the infra MCO could reboot each infra node at the same time as
+the corresponding tenant node rebooted. But this (probably) turns out
+to be unnecessary, as it should always be safe to just reboot the
+infra nodes when their tenant nodes reboot, without needing to have
+"planned" the reboot.
```

Comment:
OK this is answering my above question - infra nodes don't need to drain (or at least, the drain should be predictable and fast).

But it also seems like it's heading long term to conflict with the above goal of doing more on the DPUs.